src/AuditReader.php

```diff
@@ -254,7 +254,11 @@ public function find($className, $id, $revision, array $options = [])
         }
 
         foreach ($classMetadata->associationMappings as $assoc) {
-            if (($assoc['type'] & ClassMetadata::TO_ONE) === 0 || !$assoc['isOwningSide']) {
+            if (
+                ($assoc['type'] & ClassMetadata::TO_ONE) === 0
+                || false === $assoc['isOwningSide']
+                || !isset($assoc['joinColumnFieldNames'])
```

Comment:
why this added condition?

Reply:
The joinColumnFieldNames key might not be defined by definition, so we have to add an isset check for static analysis.

When isOwningSide is false the key is not defined. When it's true the key is defined.

But there is no dynamic understanding for phpstan saying that one value imply the fact another key exist.

So technically this check will never happen and it's just here for static analysis ; this seemed better than an assert to me.

Reply:
It's the same idea for all the other comment

Reply:
The good thing about isset is that if someday phpstan understand this thing it will be reported as always true, will it happen the same this way?

Reply:
Yes it will report the last condition as always true/false

Reply:
Then lets leave it this way!